server/src/test/java/com/feeder/server/provider/twitter/TwitterFeedProviderTest.java

```
@@ -0,0 +1,68 @@
+package com.feeder.server.provider.twitter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.AdditionalAnswers.delegatesTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.feeder.server.model.TwitterData;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import reactor.core.publisher.Flux;
+import twitter4j.ResponseList;
+import twitter4j.Status;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.User;
+
+@WebFluxTest(controllers = TwitterFeedProvider.class)
+@ExtendWith(MockitoExtension.class)
+public class TwitterFeedProviderTest {
+
+  @MockBean private Twitter mockTwitterClient;
+  @Mock private Status mockStatus;
+  @Mock private User mockUser;
+  @Autowired private TwitterFeedProvider subject;
+  private Flux<TwitterData> mockFeed;
+
+  @BeforeEach
+  public void setUp() {
+    List<Status> concreteResponseList = new ArrayList<>();
+    concreteResponseList.add(mockStatus);
+    ResponseList<Status> mockResponseList =
+        mock(ResponseList.class, delegatesTo(concreteResponseList));
+    try {
+      when(mockTwitterClient.getHomeTimeline()).thenReturn(mockResponseList);
+    } catch (TwitterException ignored) {
+    }
```

Why do we ignore the exception here?

---

getHomeTimeline() will only ever return a mock response list since its a mock twitter client. And so TwitterException would never be thrown. I could be wrong in my understanding and so, for completeness, I could add a "fail()" in the catch clause but it should never reach there anyway.

---

If we need to faciliate a checked exception being thrown here, maybe we can just make the method signature throw TwitterException? IMO is clearer to understand than the empty trycatch.